### DISCUSSION TITLE
MGCB can't build sound files

### DISCUSSION THREAD

**Discussion by AlexiyOrlov** [MonoGame/MonoGame#8217] [Audio]
I tried adding OGG and WAV files via MGCB editor, but it fails to build them.
![Screenshot 2024-03-05 151935](https://github.com/MonoGame/MonoGame/assets/5513819/469b70ae-dd6f-4e3a-b021-5ecb9e20ee2a)

**Comment by ThomasFOG** [comment]
What are the properties set on that file (importer and processor) and what is the error?

**Comment by AlexiyOrlov** [comment]
Importer is OGG importer, processor is Sound effect. The error is 'The command "dotnet mgcb /quiet /@:"D:\C# projects\MyGame\MyGame\Content\Content.mgcb" /platform:DesktopGL /outputDir:"D:/C# projects/MyGame/MyGame/Content/bin/DesktopGL/Content" /intermediateDir:"D:/C# projects/MyGame/MyGame/Content/obj/DesktopGL/net6.0/Content" /workingDir:"D:/C# projects/MyGame/MyGame/Content/"" exited with code 1' in Visual studio, if that's the right error.

**Reply by vkocjancic** [reply]
Have you managed to resolve this? Because, I am experiencing the exact same error with v3.8.3.0 of mgcb on Linux platform.
The ogg file plays in VLC player on the same machine. 

I tried running the mgcb build command without /quiet parameter and I get this output:
`/home/vkocjancic/projects/games/hw/src/hw/Content/audio/bong_001.ogg
Failed to open file bong_001. Ensure the file is a valid audio file and is not DRM protected.
`

I first thought it was a DRM issue, but then I recorded my own ogg file on the same machine and the issue persisted. Don't know what to do anymore, tbh. 

Oh, and file priviliges are set to rwxrwxrwx.

**Reply by AlexiyOrlov** [reply]
No. I left Monogame

**Comment by tomspilman** [comment]
This was the same issue i found.  

It worked for me when i ran a debug build of the pipeline...  it was some weird thing with the NuGet resolution of FFmpeg.

**Comment by vkocjancic** [comment]
I found the reason. I checked the source code and noticed that much of the pipeline for audio files just does work for WAV files. For all rest, it is just getting full file path and then trying to obtain file data by calling mgcb-ffprobe.

So I located mgcb-ffprobe and ran it and find out that it requires .NET 8 to run. Otherwise it fails spectacularly. Maybe, there is a case of doing a standalone build of mgcb-ffprobe. But I digress...
After installing latest .NET 8 SDK, mgcb-ffprobe ran without a glitch and after that, the error was gone. I hope this helps you in any way.

Oh... and I don't mean to bitch, because, MonoGame is great project and it is free, but audio pipeline needs a bit of error handling overhaul.
